lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -600,6 +600,25 @@ SILCombiner::optimizeConcatenationOfStringLiterals(ApplyInst *AI) {
   return tryToConcatenateStrings(AI, Builder);
 }
 
+/// Determine the pattern for global_addr..
+/// %3 = global_addr @$P : $*SomeP
+/// %4 = init_existential_addr %3 : $*SomeP, $SomeC
+/// %5 = alloc_ref $SomeC
+/// store %5 to %4 : $*SomeC
+static SILValue findInitExistentialFromGlobalAddr(GlobalAddrInst *GAI) {
+  /// Check for a single InitExistential.
+  SILInstruction *SingleIE = NULL;
+  for (auto Use : GAI->getUses()) {
+    auto *User = Use->getUser();
+    if(auto *InitExistential = dyn_cast<InitExistentialAddrInst>(User)) {
+      if (SingleIE) return SILValue();
+      SingleIE = InitExistential;
+    }
+  }
+  if (!SingleIE) return SILValue();
+  return cast<InitExistentialAddrInst>(SingleIE);
```

Comment:
Can you guarantee that through some sort of obfuscation there aren't any init existential addr inst. For instance, imagine if I do a cast then cast back and then do init existential addr inst. (I just made this up keep in mind).